atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/expectations/ExpectationsMeasuringUtils.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.transaction.impl.expectations;
+
+import com.google.common.collect.Multimap;
+import com.palantir.atlasdb.keyvalue.api.Cell;
+import com.palantir.atlasdb.keyvalue.api.RangeRequest;
+import com.palantir.atlasdb.keyvalue.api.RowResult;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.atlasdb.keyvalue.api.Value;
+import com.palantir.util.paging.TokenBackedBasicResultsPage;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.function.Function;
+
+public final class ExpectationsMeasuringUtils {
+    private ExpectationsMeasuringUtils() {}
+
+    public static long byteArrayByTableReferenceByteSize(Map<TableReference, byte[]> arrayByRef) {
+        return arrayByRef.entrySet().stream()
+                .mapToLong(entry -> byteSize(entry.getKey()) + entry.getValue().length)
+                .sum();
+    }
+
+    public static long longByCellByteSize(Map<Cell, Long> timestampByCell) {
+        return timestampByCell.keySet().stream()
```

Comment:
Right so since this is a util class, i would recommend not passing in something like `timestamp` in the name.
Also the names of methods are too complex. You can have just one method called `sizeInBytes` and overload.

Reply:
Will review and change the param names

For overloading, I will try to reduce the number of distinct names. My original plan was to have just one `byteSize` function but the compiler wasn't happy when you overloaded with different `Map` generic types

Will look into that more